test/javascript/components/tabs.js

```diff
@@ -1,20 +1,150 @@
 describe('Tabs', function() {
-	var plugin;
-	var $html;
-
-	// afterEach(function() {
-	// 	plugin.destroy();
-	// 	$html.remove();
-	// });
-
-	describe('constructor()', function() {
-		// it('', function() {
-		// 	$html = $('').appendTo('body');
-		// 	plugin = new Foundation.Tabs($html, {});
-
-		// 	plugin.$element.should.be.an('object');
-		// 	plugin.options.should.be.an('object');
-		// });
-	});
+  var plugin;
+  var $html;
+  var template = `
+    <div>
+      <ul class="tabs" data-tabs id="example-tabs">
+        <li class="tabs-title is-active"><a href="#panel1" aria-selected="true">Tab 1</a></li>
+        <li class="tabs-title"><a href="#panel2">Tab 2</a></li>
+        <li class="tabs-title"><a href="#panel3">Tab 3</a></li>
+      </ul>
+
+      <div class="tabs-content" data-tabs-content="example-tabs">
+        <div class="tabs-panel is-active" id="panel1">
+          <p>one</p>
+          <p>Check me out! I'm a super cool Tab panel with text content!</p>
+        </div>
+        <div class="tabs-panel" id="panel2">
+          <p>two</p>
+          <p>Check me out! I'm a super cool Tab panel with text content!</p>
+        </div>
+        <div class="tabs-panel" id="panel3">
+          <p>three</p>
+          <p>Check me out! I'm a super cool Tab panel with text content!</p>
+        </div>
+      </div>
+    </div>`;
+
+  afterEach(function() {
+    plugin.destroy();
+    $html.remove();
+  });
+
+ describe('constructor()', function() {
+    it('stores the element and plugin options', function() {
+      $html = $(template).appendTo('body');
+      plugin = new Foundation.Tabs($html.find('[data-tabs]'), {});
+
+      plugin.$element.should.be.an('object');
+      plugin.options.should.be.an('object');
+    });
+  });
+
+  describe('init()', function() {
+    it('sets ARIA attributes', function() {
+      $html = $(template).appendTo('body');
+      plugin = new Foundation.Tabs($html.find('[data-tabs]'), {});
+            
+      // Panels
+      $html.find('#panel1').should.have.attr('role', 'tabpanel');
+      $html.find('#panel1').should.have.attr('aria-labelledby', $html.find('[href="#panel1"]').attr('id'));
+      $html.find('#panel1').should.have.attr('aria-hidden', 'false');
+      $html.find('#panel2').should.have.attr('aria-hidden', 'true');
+
+      // Links
+      $html.find('[href="#panel1"]').should.have.attr('role', 'tab');
+      $html.find('[href="#panel1"]').should.have.attr('aria-controls', $html.find('panel1').attr('id'));
+      $html.find('[href="#panel1"]').should.have.attr('aria-selected', 'true');
+      $html.find('[href="#panel2"]').should.have.attr('aria-selected', 'false');
+
+      // Tab list items
+      $html.find('[href="#panel1"]').parent().should.have.attr('role', 'presentation');
+    });
+
+    // TODO: Check if classes should be applied by the Plugin or not.
```

Comment:
I don't think the plugin should apply classes; rather it looks like it takes some arguments that tell it what classes to look for.  This is relatively inline with our general policy that data-attributes should be communicating to the JS and classes should be style related.

Part of me thinks instead we should be labeling links & panels with data attributes, thus leaving the classes to be completely style related and not hunted for by the JS, similar to how off canvas works, but regardless I think the only classes that JS should be adding are state related ('is-active'), not the base classes.